solutions/usaco-1185.mdx

```diff
@@ -0,0 +1,141 @@
+---
+id: usaco-1185
+source: USACO Gold 2022 January
+title: Drought
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://usaco.org/current/data/sol_prob1_gold_jan22.html)
+
+## Explanation
+
+Let's consider a case where we have just two cows.
+
+$[2, 2]$
+
+In any case where they're equal, it's always possible to reduce them both to zero! (by decreasing adjacent elements).
+
+However, with three cows:
+
+$[2, 2, 2]$
+
+We can't apply the same transformation. In fact, for any odd number of cows, it's impossible to guarantee that we can decrement all the elements to zero.
+
+This motivates us to split the problem into two cases, if $N$ is even or if $N$ is odd.
+
+If $N$ is even, then we can count the number of tuples such that all the elements are at zero because we can guarantee that this tuple can reach all equal states and vice versa.
+
+However, if $N$ is odd, we can't make the same assumption, they could be equal at any value! Thus, we have to count the number of tuples for all possible hunger levels $x$. To account for this, note that by shifting the ceiling of the data by $x$, we can find the answer for setting all cows to $x$ with the same process as earlier.
+
+What information do we have to keep track of at point $i$?
```

Comment:
hmm i think just mentioning point i before the dp is a bit weird since it's unclear what "point i" is

Reply:
makes sense, thanks!